esrally/mechanic/launcher.py

```diff
@@ -186,14 +188,28 @@ def _set_env(self, env, k, v, separator=' ', prepend=False):
             else:
                     env[k] = env[k] + separator + v
 
+    @staticmethod
+    def _run_subprocess(command_line, env):
+        command_line_args = shlex.split(command_line)
+
+        # pylint: disable=subprocess-popen-preexec-fn
```

Comment:
I know we didn't comment this last time when this was added, but when going back over this code I completely forgot about why we have a preexec_fn.  Seems an uncommonly used function that is prone to trouble as in the Pylint warning.  Perhaps we can briefly comment why it is necessary for detached processes?

Reply:
Alternatively, we can "upgrade" to using the `start_new_session` argument described in https://docs.python.org/3.2/library/subprocess.html

Reply:
Thanks for the feedback! As this is currently blocking our nightly benchmarks I'll merge this PR as is but this is definitely a good idea for a for a future PR. I'll keep this on my radar! :)

Reply:
I've pushed https://github.com/elastic/rally/pull/884.